.github/workflows/reusable-docs.yml

```diff
@@ -23,11 +23,16 @@ jobs:
       refspec_pr: '+${{ github.event.pull_request.head.sha }}:remotes/origin/${{ github.event.pull_request.head.ref }}'
     steps:
     - name: 'Check out latest PR branch commit'
+      if: github.event_name == 'pull_request'
       uses: actions/checkout@v4
       with:
         ref: ${{ github.event.pull_request.head.sha }}
+    - name: 'Check out latest commit'
+      if: github.event_name != 'pull_request'
+      uses: actions/checkout@v4
```

Comment:
Let's move this after step 'Fetch commits to get branch diff', so instead of steps for:
```yml
      if: github.event_name == 'pull_request'
      ...
      if: github.event_name != 'pull_request'
      ...
      if: github.event_name == 'pull_request'
      ...
```

We group them:
```yml
      if: github.event_name == 'pull_request'
      ...
      if: github.event_name == 'pull_request'
      ...
      if: github.event_name != 'pull_request'
      ...
```